codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/customizations/ApiKeyAuthDecoratorTest.kt

```diff
@@ -0,0 +1,175 @@
+        }
+
+        structure SomeOutput {
+            someAttribute: Long,
+            someVal: String
+        }
+
+        @http(uri: "/SomeOperation", method: "GET")
+        operation SomeOperation {
+            output: SomeOutput
+        }
+    """.asSmithyModel()
+
+    @Test
+    fun `set an api key in query parameter`() {
+        val testDir = clientIntegrationTest(
+            modelQuery,
+            // just run integration tests
+            IntegrationTestParams(command = { "cargo test --test *".runWithWarnings(it) }),
+        ) { clientCodegenContext, rustCrate ->
+            rustCrate.integrationTest("api_key_present_in_property_bag") {
+                val moduleName = clientCodegenContext.moduleUseName()
+                Attribute.TokioTest.render(this)
+                rust(
+                    """
+                    async fn api_key_present_in_property_bag() {
+                        use aws_smithy_http_auth::api_key::AuthApiKey;
+                        let api_key_value = "some-api-key";
+                        let conf = $moduleName::Config::builder()
+                            .api_key(AuthApiKey::new(api_key_value))
+                            .build();
+                        let operation = $moduleName::operation::SomeOperation::builder()
+                            .build()
+                            .expect("input is valid")
+                            .make_operation(&conf)
+                            .await
+                            .expect("valid operation");
+                        let props = operation.properties();
+                        let api_key_config = props.get::<AuthApiKey>().expect("api key in the bag");
+                        assert_eq!(
+                            api_key_config,
+                            &AuthApiKey::new(api_key_value),
+                        );
+                    }
+                    """,
+                )
+            }
+
+            rustCrate.integrationTest("api_key_auth_is_set_in_query") {
+                val moduleName = clientCodegenContext.moduleUseName()
+                Attribute.TokioTest.render(this)
+                rust(
+                    """
+                    async fn api_key_auth_is_set_in_query() {
+                        use aws_smithy_http_auth::api_key::AuthApiKey;
+                        let api_key_value = "some-api-key";
+                        let conf = $moduleName::Config::builder()
+                            .api_key(AuthApiKey::new(api_key_value))
+                            .build();
+                        let operation = $moduleName::operation::SomeOperation::builder()
+                            .build()
+                            .expect("input is valid")
+                            .make_operation(&conf)
+                            .await
+                            .expect("valid operation");
+                        assert_eq!(
+                            op.request().uri().query(),
```

Comment:
This might fix the test compile:
```suggestion
                            operation.request().uri().query(),
```